src/adapters/methods/lora.py

```diff
@@ -432,6 +451,58 @@ def init_shared_vera_parameters(lora_A_shape, lora_B_shape, adapter_config, devi
     return parameters
 
 
+# DoRA Methods
+def compute_dora_norm(weights: torch.Tensor, added: torch.Tensor) -> torch.Tensor:
+    """This function calculates the column-wise norm of the pre-trained weights
+    and the LoRA matrices
+
+    eg: for LoRA, it would be ||w_0 + BA||c
+    """
+    return torch.linalg.norm(weights + added, dim=1)
+
+
+def compute_dora_deltaw(
+    orig_result: torch.Tensor, added: torch.Tensor, m: torch.Tensor, norm: torch.Tensor, scaling: float = 1.0
+) -> torch.Tensor:
```

Comment:
```suggestion
def compute_dora_deltaw(
    orig_result: torch.Tensor, added: torch.Tensor, m: torch.Tensor, norm: torch.Tensor
) -> torch.Tensor:
```

scaling not needed anymore